testsuite/ada_lsp/refactor_imports_commands/refactor_imports_commands.json

```diff
@@ -0,0 +1,342 @@
+                "id": 1,
+                "method": "initialize"
+            },
+            "wait": [
+                {
+                    "id": 1,
+                    "result": {
+                        "capabilities": {
+                            "textDocumentSync": 2,
+                            "completionProvider": {
+                                "triggerCharacters": [
+                                    ".",
+                                    "("
+                                ],
+                                "resolveProvider": false
+                            },
+                            "hoverProvider": true,
+                            "declarationProvider": true,
+                            "definitionProvider": true,
+                            "typeDefinitionProvider": true,
+                            "implementationProvider": true,
+                            "referencesProvider": true,
+                            "documentSymbolProvider": true,
+                            "codeActionProvider": {},
+                            "documentFormattingProvider": true,
+                            "renameProvider": {},
+                            "foldingRangeProvider": true,
+                            "executeCommandProvider": {
+                                "commands": [
+                                    "als-named-parameters",
+                                    "als-refactor-imports"
+                                ]
+                            },
+                            "workspaceSymbolProvider": true,
+                            "alsCalledByProvider": true,
+                            "alsCallsProvider": true,
+                            "alsShowDepsProvider": true,
+                            "alsReferenceKinds": [
+                                "reference",
+                                "access",
+                                "write",
+                                "call",
+                                "dispatching call",
+                                "parent",
+                                "child"
+                            ]
+                        }
+                    }
+                }
+            ]
+        }
+    },
+    {
+        "send": {
+            "request": {
+                "jsonrpc": "2.0",
+                "method": "initialized"
+            },
+            "wait": []
+        }
+    },
+    {
+        "send": {
+            "request": {
+                "params": {
+                    "settings": {
+                        "ada": {
+                            "projectFile": "default.gpr",
+                            "scenarioVariables": {},
+                            "defaultCharset": "ISO-8859-1"
+                        }
+                    }
+                },
+                "jsonrpc": "2.0",
+                "method": "workspace/didChangeConfiguration"
+            },
+            "wait": []
```

Comment:
I suggest you to add this:

            "wait": [
                {
                    "jsonrpc": "2.0",
                    "id": 2,
                    "method": "window/workDoneProgress/create",
                    "params": {
                        "token": "<ANY>"
                    }
                },  {
                    "jsonrpc": "2.0",
                    "method": "$/progress",
                    "params": {
                      "token": "<ANY>",
                      "value": {
                        "kind": "end"
                      }
                    }
                }
            ]
        }

This will wait until ALS finish the indexing, so you will be sure there is no other indexing requests from the ALS.

Reply:
Agree. Exactly what I was looking for.